### PR TITLE
feat(content-answers): Disable button when not allowed file type

### PR DESCRIPTION
**Pull request by greg-in-a-box**
Ignore the fact that the button looks the way it does, its being fixed in a different PR

image

### CODE REVIEW

File: src/elements/content-preview/ContentPreview.js [hunk 1]
```diff
@@ -1323,7 +1323,12 @@ class ContentPreview extends React.PureComponent<Props, State> {
                             canDownload={this.canDownload()}
                             canPrint={canPrint}
                             onDownload={this.download}
-                            contentAnswersProps={contentAnswersProps}
+                            contentAnswersProps={{
+                                ...contentAnswersProps,
+                                fileId: currentFileId,
+                                fileExtension: currentExtension,
+                                versionId: selectedVersionId,
```
Comment (tjuanitas) [inline src/elements/content-preview/ContentPreview.js]: these are also set in `PreviewHeader`:
https://github.com/box/box-ui-elements/blob/c987daea3de9e44eb54f363641750215bbf08613/src/elements/content-preview/preview-header/PreviewHeader.js#L66-L68

so no need to pass from this component

Comment (greg-in-a-box) [reply src/elements/content-preview/ContentPreview.js]: I literally removed this before you made the comment

File: src/features/content-answers/ContentAnswers.tsx [hunk 2]
```diff
@@ -15,10 +21,10 @@ const ContentAnswers = () => {
...
     return (
         <div>
-            <ContentAnswersOpenButton onClick={handleClick} />
+            <ContentAnswersOpenButton onClick={handleClick} fileExtension={fileExtension} />
         </div>
     );
 };
 
-export type ContentAnswersProps = ExternalProps;
+export type ContentAnswersProps = ExternalProps & Props;
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswers.tsx:29]: is this change still needed for how we're currently using `ContentAnswersProps`?

File: src/features/content-answers/ContentAnswersOpenButton.tsx [hunk 3]
```diff
@@ -12,12 +13,22 @@ import './ContentAnswersOpenButton.scss';
...
     const { formatMessage } = intl;
 
+    const isAllowedFileType = (extension: string) => {
+        const allowedTypes = CODE_FILE_EXTENSIONS.concat(DOCUMENT_FILE_EXTENSIONS).concat(TEXT_FILE_EXTENSIONS);
+        return allowedTypes.indexOf(extension) !== -1;
+    };
+
     const getTooltipText = () => {
+        if (!isAllowedFileType(fileExtension)) {
+            return intl.formatMessage(messages.disabledTooltipFileNotCompatible);
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswersOpenButton.tsx:29]:
```suggestion
            return formatMessage(messages.disabledTooltipFileNotCompatible);
```

File: src/features/content-answers/ContentAnswersOpenButton.tsx [hunk 4]
```diff
@@ -12,12 +13,22 @@ import './ContentAnswersOpenButton.scss';
...
+    fileExtension: string;
 }
 
-const ContentAnswersOpenButton = ({ intl, onClick }: ContentAnswersOpenButtonProps) => {
+const ContentAnswersOpenButton = ({ intl, onClick, fileExtension }: ContentAnswersOpenButtonProps) => {
     const { formatMessage } = intl;
 
+    const isAllowedFileType = (extension: string) => {
+        const allowedTypes = CODE_FILE_EXTENSIONS.concat(DOCUMENT_FILE_EXTENSIONS).concat(TEXT_FILE_EXTENSIONS);
+        return allowedTypes.indexOf(extension) !== -1;
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswersOpenButton.tsx:24]: .
```suggestion
        return allowedTypes.includes(extension);
```

File: src/features/content-answers/ContentAnswersOpenButton.tsx [hunk 5]
```diff
@@ -12,12 +13,22 @@ import './ContentAnswersOpenButton.scss';
 interface ContentAnswersOpenButtonProps {
     intl: IntlShape;
     onClick: () => void;
+    fileExtension: string;
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswersOpenButton.tsx:16]: 🔤

Comment (tjuanitas) [reply src/features/content-answers/ContentAnswersOpenButton.tsx:16]: same for a few places below:
~line 19
~line 42

**Review by tjuanitas — COMMENTED** [review]
quick thoughts! I can do a more thorough review soon

**Comment by greg-in-a-box** [comment]
> quick thoughts! I can do a more thorough review soon

fixed

File: src/features/content-answers/ContentAnswersOpenButton.tsx [hunk 6]
```diff
@@ -12,12 +13,22 @@ import './ContentAnswersOpenButton.scss';
...
 }
 
-const ContentAnswersOpenButton = ({ intl, onClick }: ContentAnswersOpenButtonProps) => {
+const ContentAnswersOpenButton = ({ intl, onClick, fileExtension }: ContentAnswersOpenButtonProps) => {
     const { formatMessage } = intl;
 
+    const isAllowedFileType = (extension: string) => {
+        const allowedTypes = CODE_FILE_EXTENSIONS.concat(DOCUMENT_FILE_EXTENSIONS).concat(TEXT_FILE_EXTENSIONS);
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswersOpenButton.tsx:23]: curious your thoughts on this:
```
        const allowedTypes = [
            ...CODE_FILE_EXTENSIONS,
            ...DOCUMENT_FILE_EXTENSIONS,
            ...TEXT_FILE_EXTENSIONS,
        ];
```

I've seen spread operator more common than `concat` in modern js but not sure if there's an objective reasons for using one over the other. thoughts?

Comment (greg-in-a-box) [reply src/features/content-answers/ContentAnswersOpenButton.tsx:23]: im fine with changing it, this was copied over

File: src/elements/content-preview/stories/ContentPreview.stories.js [hunk 7]
```diff
@@ -87,6 +87,26 @@ export const PreviewWithSidebar = () => {
...
 
+export const PreviewWithBoxAI = () => {
+    const fileId = text('File ID', global.FILE_ID);
+    const token = text('Access Token', global.TOKEN);
+
+    return (
+        <IntlProvider locale="en">
+            <ContentPreview
+                contentAnswersProps={{
+                    show: true,
```
Comment (tjuanitas) [inline src/elements/content-preview/stories/ContentPreview.stories.js:98]: do we want to enable this yet in storybook since we're still implementing?

File: src/features/content-answers/constants.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    'ppt',
+    'pptx',
+    'rtf',
+    'wpd',
+    'xls',
+    'xlsm',
+    'xlsx',
+    'key',
+    'pages',
+    'numbers',
```
Comment (tjuanitas) [inline src/features/content-answers/constants.ts:78]: nit: most of this list is already alpha sorted

File: src/elements/content-preview/stories/ContentPreview.stories.js (hunk 7 above)
Comment (greg-in-a-box) [reply src/elements/content-preview/stories/ContentPreview.stories.js:98]: This is the easiest way to test the feature itself, I dont see much of an issue since people can see the PRs go through and we dont advertise storybook on the docs

File: src/elements/content-preview/preview-header/PreviewHeader.js [hunk 9]
```diff
@@ -100,7 +101,13 @@ const PreviewHeader = ({
                                     {...contentOpenWithProps}
                                 />
                             )}
-                            {shouldRenderAnswers && <ContentAnswers />}
+                            {shouldRenderAnswers && (
+                                <ContentAnswers
+                                    fileExtension={currentExtension}
```
Comment (karelee7) [inline src/elements/content-preview/preview-header/PreviewHeader.js:106]: Are there any tests for PreviewHeader that we need to update after adding these props?

File: src/features/content-answers/ContentAnswers.tsx [hunk 10]
```diff
@@ -4,10 +4,13 @@ import ContentAnswersModal from './ContentAnswersModal';
 import ContentAnswersOpenButton from './ContentAnswersOpenButton';
 
 type ExternalProps = {
```
Comment (tjuanitas) [inline src/features/content-answers/ContentAnswers.tsx:6]: `ExternalProps` is used by the customer to pass specific props to configure ContentAnswers: https://github.com/box/box-ui-elements/blob/master/src/elements/content-preview/ContentPreview.js#L233

the new props like `fileExtension`, etc. can't be used by customer

File: src/elements/content-preview/preview-header/PreviewHeader.js (hunk 9 above)
Comment (greg-in-a-box) [reply src/elements/content-preview/preview-header/PreviewHeader.js:106]: updated

File: src/features/content-answers/ContentAnswers.tsx (hunk 10 above)
Comment (greg-in-a-box) [reply src/features/content-answers/ContentAnswers.tsx]: changed

**Review by tjuanitas — APPROVED** [review]

**Review by karelee7 — APPROVED** [review]
Looks good!
